### PR TITLE
Replace issue templates with issue forms

### LINKED ISSUE #884: Issue forms
### Summary
Issue forms entered open beta a few months ago, and it would be good to switch to them.

### Problem
This would fix problems with people submitting largely undescriptive issues and PRs after having deleted the markdown template. Issue forms do not have this same problem.

### Ideal implementation
The standard way recommended by GitHub. I plan to do this myself, so I'll port what you currently have over.

### Checklist
- [x] I have searched the issue tracker and have made sure it's not a duplicate.
      If it is a follow up of another issue, I have specified it.

### PR DESCRIPTION
**Pull request by parafoxia**
### Summary
Adds issue forms.

### Checklist
- [ ] ~~I have run `nox` and all the pipelines have passed.~~ Not necessary
- [ ] ~~I have made unittests according to the code I have added/modified/deleted.~~ Not necessary

### Related issues
Closes #884.

### CODE REVIEW

**Comment by parafoxia** [comment]
I'm not sure if the fragments check needs to pass for this as there are no changes in the API itself, but let me know if you want me to add something.

**Review by davfsa — APPROVED** [review]
I actually had a stashed half done version of this, but ig I can delete that now. Thanks for taking the time to work on this :+1:

**Comment by davfsa** [comment]
For context, this is what they would look like:
image
image

File: .github/ISSUE_TEMPLATE/feature_request.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    id: problem
+    attributes:
+      label: Problem
+      description: What problem will this feature solve (if any)?
+
+  - type: textarea
+    id: ideal-implementation
+    attributes:
+      label: Ideal implementation
+      description: How should this feature be implemented?
```
Comment (davfsa) [inline .github/ISSUE_TEMPLATE/feature_request.yml:24]:
```suggestion
      description: How should this feature be implemented?
      value: To be decided
```

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    id: system-info
+    attributes:
+      label: System info
+      description: Output of `python -m hikari`.
+      placeholder: |
+        hikari (2.0.0.dev103) [dc0f905a]
+        located at /home/username/my-bot/.venv/lib/python3.10/site-packages/hikari
+        CPython 3.10.0 GCC 10.2.1 20210110
+        Linux Skye 5.4.72-microsoft-standard-WSL2 #1 SMP Wed Oct 28 23:40:43 UTC 2020 x86_64
+    validations:
```
Comment (davfsa) [inline .github/ISSUE_TEMPLATE/bug_report.yml:45]:
```suggestion
    render: shell
    validations:
```

File: .github/ISSUE_TEMPLATE/feature_request.yml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+      label: Summary
+      description: Small summary of the feature.
+    validations:
+      required: true
+
+  - type: textarea
+    id: problem
+    attributes:
+      label: Problem
+      description: What problem will this feature solve (if any)?
```
Comment (davfsa) [inline .github/ISSUE_TEMPLATE/feature_request.yml:18]: This section could probably be changed to something like "Why is this needed?"
```suggestion
  - type: textarea
    id: problem
    attributes:
      label: Why is this needed?
      description: Why should this feature be implemented? What problem(s) would it solve?
```

Additionally, should this be marked as required?

**Review by davfsa — COMMENTED** [review]
Just small comments I would like to port from what I had stashed

**Comment by parafoxia** [comment]
Cool beans, I'll bundle those in now. And no worries haha -- I'd been meaning to learn them for some time, so when I did I thought I'd get some more practise in by porting them here lmao.

File: .github/ISSUE_TEMPLATE/feature_request.yml (hunk 3 above)
Comment (parafoxia) [reply .github/ISSUE_TEMPLATE/feature_request.yml:18]: I think in the new context, as it's now asking why the feature is needed rather than asking if it solves any problems, I think it should be. That way contributors can justify their feature before anyone has to enquire about it.

Comment (davfsa) [reply .github/ISSUE_TEMPLATE/feature_request.yml]: Ok, then required it is 👍

**Review by davfsa — APPROVED** [review]

**Review by FasterSpeeding — APPROVED** [review]
